assets/js/app.js

```diff
@@ -0,0 +1,277 @@
+"use strict";
+
+(() => {
+
+    const getData = url => fetch(url).then(response => response.json())
+
+    const fetchRetry = async (url) => {
+        let isSuccess = false
+        do {
+            try {
+                const data = await getData(url)
+                isSuccess = true
+                return data
+            } catch (e) {
+                setTimeout(() => {
+                    fetchRetry(url)
+                }, 5000)
+            }
+        } while (!isSuccess)
+    }
+
+    const getAllCoins = async () => getData('https://api.coingecko.com/api/v3/coins/list')
+    // const getAllCoins = async () => getData('assets/json/coins.json')
+    const getSingleCoin = async (coin) => fetchRetry(`https://api.coingecko.com/api/v3/coins/${coin}`)
+    const getGraphData = async (coins) => getData(`https://min-api.cryptocompare.com/data/pricemulti?fsyms=${coins.join()}&tsyms=USD`)
+
+    //set saving database on local storage, if there isn`t
+    if (!localStorage.getItem(`coinsCache`)) localStorage.setItem(`coinsCache`, JSON.stringify([]))
+    if (!localStorage.getItem(`5coinsInStorage`)) localStorage.setItem(`5coinsInStorage`, JSON.stringify([]))
+
+
+    //the coins page
+    document.getElementById("coins-click-tab").addEventListener(`click`, async () => {
```

Comment:
so the user lands on an empty page and only on re-entry to this tab the coins list is loaded (but not on startup)